component/src/test/java/org/wso2/extension/siddhi/gpl/execution/streamingml/clustering/clustree/ClusTreeStreamProcessorExtensionTest.java

```diff
@@ -550,6 +552,130 @@ public void testClusTree2D_22() throws Exception {
         }
     }
 
+    @Test
+    public void testClusTree2D_23() throws Exception {
+        logger.info("ClusTreeStreamProcessorExtension Test - Test case with non existing stream");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        //String inputStream = "define stream InputStream (x double, y double);";
+
+        String query = (
+                "@info(name = 'query1') " +
+                        "from InputStream#streamingml:clusTree(2, 10, 20, 2, -50, x, y) " +
+                        "select closestCentroidCoordinate1, closestCentroidCoordinate2, x, y " +
+                        "insert into OutputStream;");
+        try {
+            SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(query);
+        } catch (Exception e) {
+            logger.info("Error caught");
+            AssertJUnit.assertTrue(e instanceof SiddhiAppCreationException);
+            AssertJUnit.assertTrue(e.getMessage().contains("definition with ID 'InputStream' has not been defined"));
+        }
+    }
+
+    @Test
+    public void testClusTree2D_24() throws Exception {
+        logger.info("ClusTreeStreamProcessorExtension Test - Test case with non-numeric event data");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String inputStream = "define stream InputStream (x double, y double);";
+
+        String query = (
+                "@info(name = 'query1') " +
+                        "from InputStream#streamingml:clusTree(2, x, y) " +
+                        "select closestCentroidCoordinate1, closestCentroidCoordinate2, x, y " +
+                        "insert into OutputStream;");
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(inputStream + query);
+
+        siddhiAppRuntime.addCallback("query1", new QueryCallback() {
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(inEvents);
+            }
+        });
+
+
+        siddhiAppRuntime.start();
```

Comment:
Shutdown the siddhiAppRuntime and SiddhiManager after executing the test case

Reply:
Fixed in https://github.com/wso2-extensions/siddhi-gpl-execution-streamingml/pull/29